arrays/arrays.c

```diff
@@ -83,14 +99,22 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+  if (index > arr->count) {
+    fprintf(stderr, "INSERT error: %d", index);
+  }
   // Resize the array if the number of elements is over capacity
-
+  if (arr->count + 1 > arr->capacity) { 
```

Comment:
Remember the count could be at or above the capacity before you even insert. You want to resize as soon as the count is _"at"_ capacity... but it could also possibly be over capacity. Think about how that would change your conditional here.